src/app/components/TaskFormUpdate.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState, FormEvent } from "react";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { toast } from "react-toastify";
+import { Task } from "../types/task";
+
+const Categories_List = ["choose", "Work", "Personal", "Health", "Other"];
+
+interface TaskUpdateFormProps {
+  task: Task;
+  updateTask: (formData: FormData, id: string) => Promise<void>;
+  onClose: () => void;
+}
+
+export default function TaskUpdateForm({
+  task,
+  updateTask,
+  onClose,
+}: TaskUpdateFormProps) {
+  const [updatedTask, setUpdatedTask] = useState(task);
+  updatedTask.date = new Date(task.date).toISOString().split("T")[0];
+
+  async function onSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+    formData.append("userId", task.userId);
+    try {
+      await updateTask(formData, task.id);
```

Comment:
**Question: can you pull updated tasks from the server after updating or blow the cache ( which sort of does the ssame thing)**

Reply:
yes I used it, I got from server using mutate